02/app.js

```diff
@@ -2,12 +2,26 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 
 class Counter extends React.Component {
-    state = {
-        amount: 0,
+    
+    constructor(props) {
+        super(props);
+        this.state = {
+            amount: 0,
+        }
+
+        this.clickCounter = this.clickCounter.bind(this);
+    }
+
+    clickCounter() {
+        const { amount } = this.state;
+
+        this.setState({
+            amount: amount + 1,
+        })
     }
     
     render() {
-        return <button>click me ({ this.state.amount })</button>
+        return <button onClick={ this.clickCounter }>click me ({ this.state.amount })</button>
```

Comment:
Można też użyć funkcji strzałkowej i wtedy bind nie będzie potrzebny tj. `onClick={ () => this.clickCounter()}`